### PR TITLE
Automate Uploading to Pypi

### PR DESCRIPTION
**Pull request by coldenate**
# Automate Uploading to PyPI

In this pr, I implemented a Github Action workflow that installs the project's dependencies, builds a release, and then publishes that release to PyPI. 

There will be some setup required on your end, @TechWiz-3, but I am happy to walk you through it! My discord is `colderinit#8448`. 

The PyPI account Token is required through Github Actions Secrets. `${{ secrets.PYPI_TOKEN }}`
image

### CODE REVIEW

**Comment by coldenate** [comment]
Fixing #2.

File: .gitignore [hunk 1]
```diff
@@ -159,9 +159,10 @@ cython_debug/
...
 #.idea/
 
-
 # Vim swap files
 *.swp
 .*.swp
 *~
 .*~
+.vscode/settings.json
+poetry.lock
```
Comment (aaaandrzej) [inline .gitignore]: Are you sure you don't want to push `poetry.lock` to the repo? See poetry docs: https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control

Comment (coldenate) [reply .gitignore]: Ooh. Yeah, I think you're right. Will do!

**Comment by aaaandrzej** [comment]
I'm not a maintainer here so I'll leave the further review to the repo
maintainers.

**Comment by TechWiz-3** [comment]
Thanks a lot for your work, just wondering why did you choose `poetry` instead of `setup.py` and twine?

**Comment by coldenate** [comment]
By using 'pyproject.toml' and 'poetry' we can retain the project's core metadata in a human-readable yet machine-readable form. 
Using toml files has also become the standard as of [pep621](https://peps.python.org/pep-0621/)!

**Comment by TechWiz-3** [comment]
Thanks a lot for your work, the explanation and your followup, really appreciate that 🙏

**Pull request merged**